core/src/main/java/org/elasticsearch/ElasticsearchTimeoutException.java

```diff
@@ -33,6 +33,10 @@ public ElasticsearchTimeoutException(StreamInput in) throws IOException {
         super(in);
     }
 
+    public ElasticsearchTimeoutException(Throwable cause) {
+        super(cause);
+    }
+
     public ElasticsearchTimeoutException(String message, Object... args) {
         super(message);
     }
```

Comment:
This is not related to your changes (can be followup) but is an obvious bug (just discards `args` completely).